### PR TITLE
Dev/143 Create Admin Entity

### PR DESCRIPTION
**Pull request by Chaoyuuu**
# Description

See the linked issues for the details.

- If you have other implementation details, write it down here.

### CODE REVIEW

**Review by Wally5077 — APPROVED** [review]
LGTM

File: Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/StudentControllerIT.java [hunk 1]
```diff
@@ -128,6 +126,30 @@ private ResultActions signUp(String name, String email, String password) throws
...
+    @Test
+    void GivenOneAdminSignedUp_WhenAdminLoginCorrectly_ShouldRespondLoginResponseWithCorrectToken() throws Exception {
+        StudentView studentView = signUpAndGetResponseBody(admin);
+        LoginResponse body = signInAndGetResponseBody(this.admin.getEmail(), this.admin.getPassword());
+
+        assertEquals(studentView.id, body.id);
+        assertEquals(studentView.email, body.email);
+        TokenService.Token token = tokenService.parseAndValidate(body.token);
+        assertEquals(studentView.id, token.getStudentId());
+        assertTrue(body.isAdmin);
```
Comment (Johnny850807) [inline Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/StudentControllerIT.java:150]: should extract the duplicate codes into a method `testStudentSignUp(:Student)`

File: Student-Service/src/main/java/tw/waterball/judgegirl/studentservice/domain/usecases/SignUpUseCase.java [hunk 2]
```diff
@@ -36,11 +37,19 @@ public void execute(Request request, Presenter presenter) {
...
         student.validate();
         presenter.setStudent(studentRepository.save(student));
     }
 
+    private Student createStudent(Request request) {
+        if (request.isAdmin) {
+            return new Admin(request.name, request.email, request.password);
+        } else {
+            return new Student(request.name, request.email, request.password);
+        }
```
Comment (Johnny850807) [inline Student-Service/src/main/java/tw/waterball/judgegirl/studentservice/domain/usecases/SignUpUseCase.java:50]: return `request.isAdmin ? new Admin(request.name, request.email, request.password) : new Student(request.name, request.email, request.password)`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done.

**Review by Johnny850807 — APPROVED** [review]
LGTM!

**Pull request merged**